File: EcoQuestAPI/src/main/java/com/proyecto/spring/dto/UsuarioDTO.java
package com.proyecto.spring.dto;

import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.proyecto.spring.modelos.Evento;
import java.util.List;
import java.util.stream.Collectors;
import com.proyecto.spring.modelos.Usuario;

@JsonPropertyOrder({"id", "nombreUsuario", "nombre", "apellido", "descripcion", "edad", "email", "imagen", "comunidades", "eventos"})
public class UsuarioDTO {
    public Long id;
    public String nombreUsuario;
    public String nombre;
    public String apellido;
    public String descripcion;
    public int edad;
    public String email;
    public String imagen;
    public List<ComunidadResumenDTO> comunidades;
    public List<Evento> eventos;

    public UsuarioDTO(Usuario u) {
        this.id = u.getId();
        this.nombreUsuario = u.getNombreUsuario();
        this.nombre = u.getNombre();
        this.apellido = u.getApellido();
        this.descripcion = u.getDescripcion();
        this.edad = u.getEdad();
        this.email = u.getEmail();
        this.imagen = u.getImagen();
        this.eventos = u.getEventos();
        this.comunidades = u.getComunidades() == null ? List.of() :
            u.getComunidades().stream()
                .map(uc -> new ComunidadResumenDTO(uc.getComunidad().getNombre(), uc.getRol()))
                .collect(Collectors.toList());
    }
}

File: frontend-android/PrototipadoComunidades/proyectobase2425/app/src/main/java/com/pmdm/proyectobase2425/dto/UsuarioDto.kt
package com.pmdm.proyectobase2425.data.dto

import com.google.gson.annotations.SerializedName

// contraseña no aparece — la API no la expone en UsuarioDTO.
data class UsuarioDto(
    @SerializedName("id")           val id: Long = 0,
    @SerializedName("nombreUsuario") val nombreUsuario: String = "",
    @SerializedName("nombre")       val nombre: String = "",
    @SerializedName("apellido")     val apellido: String = "",
    @SerializedName("descripcion")  val descripcion: String = "",
    @SerializedName("edad")         val edad: Int = 0,
    @SerializedName("email")        val email: String = "",
    @SerializedName("imagen")       val imagen: String = "",
    @SerializedName("comunidades")  val comunidades: List<ComunidadResumenDto> = emptyList(),
    @SerializedName("eventos")      val eventos: List<EventoDto> = emptyList()
)
